arch/arm/core/aarch64/mmu/arm_mmu.c

```diff
@@ -218,8 +143,13 @@ static void set_pte_block_desc(uint64_t *pte, uint64_t addr_pa,
 	case MT_NORMAL_NC:
 	case MT_NORMAL:
 		/* Make Normal RW memory as execute never */
-		if ((attrs & MT_RW) || (attrs & MT_EXECUTE_NEVER))
+		if ((attrs & MT_RW) || (attrs & MT_P_EXECUTE_NEVER))
 			desc |= PTE_BLOCK_DESC_PXN;
+
+		if (((attrs & MT_RW) && (attrs & MT_RW_AP_ELx)) ||
+		     (attrs & MT_U_EXECUTE_NEVER))
+			desc |= PTE_BLOCK_DESC_UXN;
```

Comment:
This may not be true for user space, having RW as execute never. I am not sure. Considering that this EL0 extension should not impact existing functionality, will let it go to unblock PR.